run-python/util.go

```diff
@@ -0,0 +1,35 @@
+package runner
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+)
+
+// lookupAbsolutePath looks for an executable in PATH, and returns an absolute
+// path to the found executable.
+func lookupAbsolutePath(file string) (string, error) {
+	executable, err := exec.LookPath(file)
+	if err != nil {
+		return "", err
+	}
+
+	executable, err = filepath.Abs(executable)
+	if err != nil {
+		return "", err
+	}
+
+	return executable, nil
+}
+
+// fileExists checks if a file exists and is readable by performing an os.Stat()
+// operation on the file.
+func fileExists(filename string) bool {
+	_, err := os.Stat(filename)
```

Comment:
This could return an error on permissions, perhaps you don't care, but then the function should be renamed: for completeness it should have an explicit check: `os.IsNotExist(err)`

Reply:
For the `.ready` marker `os.IsNotExist(err)` would be enough, as we don't need to read the file, but for the Python executable, not being able to read it is something we'd want to flag early. I can limit this to an existence check. It's not that important in the places where it is used, I was just trying to keep the code leaner, I guess...

Will change to the explicit check.